frontend/app/src/types/user.ts

```diff
@@ -15,13 +21,23 @@ const OtherSettings = z.object({
   krakenAccountType: KrakenAccountType.optional(),
   frontendSettings: z.string().transform(arg => {
     const data = arg ? axiosCamelCaseTransformer(JSON.parse(arg)) : {};
+    if (data[REFRESH_PERIOD] > Constraints.MAX_MINUTES_DELAY) {
+      data[REFRESH_PERIOD] = Constraints.MAX_MINUTES_DELAY;
+    }
+    if (data[QUERY_PERIOD] > Constraints.MAX_SECONDS_DELAY) {
+      data[QUERY_PERIOD] = Constraints.MAX_SECONDS_DELAY;
+    }
+    if (data[VERSION_UPDATE_CHECK_FREQUENCY] > Constraints.MAX_HOURS_DELAY) {
+      data[VERSION_UPDATE_CHECK_FREQUENCY] = Constraints.MAX_HOURS_DELAY;
+    }
```

Comment:
Out of curiosity didn't zod's preprocess work for the validation? was there a problem using that approach?

Reply:
I thought preprocess is for another purpose, but after I read it again, it should can be used. Trying it now...

Reply:
You can also use it for other purposes too, but in general, it is for preprocessing the data before passing it to the parser, which is what we need to do in this case.

Reply:
This is are achieved using preprocess now @kelsos